botocore/args.py

```diff
@@ -810,6 +818,23 @@ def _compute_checksum_config(self, config_kwargs):
             valid_options=VALID_RESPONSE_CHECKSUM_VALIDATION_CONFIG,
         )
 
+    def _resolve_protocol(self, service_model):
+        # We need to ensure `protocols` exists in the metadata before attempting to
+        #  access it directly since referencing service_model.protocols directly will
+        #  raise an UndefinedModelAttributeError if protocols is not defined
+        if service_model.metadata.get('protocols'):
```

Comment:
I'm assuming this has been done but before we release, do we know these are all identical (`protocol` v `protocols`) for the currently released models?

Reply:
That's true- for all models that currently have the `protocols` trait.  Every service with the trait also only has a length of one for the moment, so no customers should see any change in behavior from this change today.